### PR TITLE
Convert search-admin app to using components

### PR DESCRIPTION
**Pull request by kr8n3r**
This is quite a big change - I recommend reviewing commit by commit to break it up into smaller chunks.

### Queries list view - before and after
queries-main

Screenshot 2020-02-19 at 14 17 42

### Query view and add bests - before and after
query-view

search-admin integration publishing service gov uk_queries_962

### New query form - before and after
queries-form

Screenshot 2020-02-19 at 14 19 02

### External links list view - before and after
external-links-main

Screenshot 2020-02-19 at 14 22 57

### New external link form - before and after
external-links-new

search-admin integration publishing service gov uk_recommended-links

### Search results - before and after
search-results-main

Screenshot 2020-02-19 at 14 16 21

### Similar search results - before and after
similar-search-results-main

Screenshot 2020-02-19 at 14 25 37

[Trello ticket](https://trello.com/c/endkE2CK/1393-update-search-admin-ui-with-components)

### CODE REVIEW

File: app/views/queries/show.html.erb [hunk 1]
```diff
@@ -1,71 +1,86 @@
...
+  breadcrumbs: [
+    {
+      title: "Queries",
+      url: queries_path
+    },
+    {
+      title: "Queries"
+    }
+  ]
+} %>
```
Comment (bilbof) [inline app/views/queries/show.html.erb]: It looks like there's some duplication - could this be put into a shared view?

File: Gemfile [hunk 2]
```diff
@@ -5,14 +5,15 @@ gem "rails", "6.0.2.1"
 gem "bootsnap", "~> 1.1", ">= 1.1.7"
 gem "generic_form_builder", "~> 0.13.1"
 gem "mysql2", "~> 0.4.5"
-gem "sass-rails", "~> 6.0"
+gem "sass-rails", "< 6"
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply Gemfile]: this is the same issue as https://github.com/alphagov/finder-frontend/pull/1953

File: app/views/queries/_form.html.erb [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
 
-  <div class="form-group">
-    <%= f.select :match_type, Query::MATCH_TYPES.map { |t| [t.humanize, t] }, class: "form-control input-lg" %>
-  </div>
+  <%
+    match_types = Query::MATCH_TYPES.map do |t|
+      {
+        text: t.humanize,
+        value: t,
+        selected: (true if t.humanize.present? && request.path.include?(t.humanize))
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/views/queries/_form.html.erb]: unfortunately that always produces `false`

File: app/assets/javascripts/filter-table.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+(function() {
+    window.GOVUK = window.GOVUK || {};
+  
+    window.GOVUK.FilterTable = function filterList(searchTerm) {
+      var itemsToFilter = document.querySelectorAll('.govuk-table__body .govuk-table__row');
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/assets/javascripts/filter-table.js]: updated

File: app/assets/javascripts/filter-table.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    };
+  
+    var formElement = document.querySelector('[data-module=filterable-table]');
+  
+    if (formElement) {
+      var searchField = formElement.querySelector('input[name=table-filter]');
+  
+      searchField.addEventListener('input', function(e) {
+        var searchTerm = searchField.value;
+        window.GOVUK.FilterTable(searchTerm);
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/assets/javascripts/filter-table.js]: fixed

File: app/views/bets/_form.html.erb [hunk 6]
```diff
@@ -1,21 +1,47 @@
...
-  <div class="form-group">
-    <%= f.check_box :is_worst, label: 'Is worst bet?' %>
-  </div>
+  <%= render "govuk_publishing_components/components/checkboxes", {
+    name: "bet[is_worst]",
+    items: [
+      {
+        label: "Is worst bet?",
+        value: 1,
+        checked: (true unless bet.is_best)
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/views/bets/_form.html.erb]: yep 👍

File: app/views/queries/show.html.erb (hunk 1 above)
Comment (kr8n3r) [reply app/views/queries/show.html.erb]: new partial introduced

File: app/views/queries/index.html.erb [hunk 7]
```diff
@@ -1,33 +1,71 @@
...
-      <%= render 'queries/query_item', query: query %>
+  <%= GovukPublishingComponents::AppHelpers::TableHelper.helper(self) do |t| %>
+    <%= t.head do %>
+      <%= t.header "Term" %>
+      <%= t.header "Match" %>
+      <%= t.header "Bets" %>
+    <% end %>
+    <%= t.body do %>
+      <% @queries.each do |query| %>
+        <% bets = capture do %>
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/views/queries/index.html.erb]: if i do that I won't be able to use the table helper
Screenshot 2020-02-21 at 15 25 34

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -1,2 +1,23 @@
...
+    items << { text: "Queries", href: queries_path, active: is_current?(queries_path) }
+    items << { text: "External links", href: recommended_links_path, active: is_current?(recommended_links_path) }
+    items << { text: "Search results", href: results_path, active: is_current?(results_path) }
+    items << { text: "Similar search results", href: new_similar_search_result_path, active: is_current?(new_similar_search_result_path) }
+
+    items << { text: current_user.name, href: Plek.new.external_url_for("signon") }
+    items << { text: "Sign out", href: "/auth/gds/sign_out" }
+
+    items
+  end
```
Comment (bilbof) [inline app/helpers/application_helper.rb]: You could use an array here:

```
[
    { text: ... },
    { text: ... },
    ...
]
```

**Comment by bilbof** [comment]
🌮 🎉

**Pull request merged**